libs/SalesforceReact/SalesforceReact/Classes/SFSmartStoreReactBridge.m

```diff
@@ -67,31 +67,41 @@ @implementation SFSmartStoreReactBridge
 {
     NSString *soupName = [argsDict nonNullObjectForKey:kSoupNameArg];
     [self log:SFLogLevelDebug format:@"soupExists with soup name '%@'.", soupName];
-    
     BOOL exists = [[self getStoreInst:argsDict] soupExists:soupName];
     callback(@[[NSNull null],  exists ? @YES : @NO]);
 }
 
 RCT_EXPORT_METHOD(registerSoup:(NSDictionary *)argsDict callback:(RCTResponseSenderBlock)callback)
 {
-    NSString *soupName = [argsDict nonNullObjectForKey:kSoupNameArg];
+    SFSmartStore *smartStore = [self getStoreInst:argsDict];
+    NSDictionary *soupSpecDict = [argsDict nonNullObjectForKey:kSoupSpecArg];
+    SFSoupSpec *soupSpec = nil;
+    NSString *soupName = nil;
+    if (soupSpecDict) {
+        soupSpec = [SFSoupSpec newSoupSpecWithDictionary:soupSpecDict];
+    } else {
+        soupName = [argsDict nonNullObjectForKey:kSoupNameArg];
+        soupSpec = [SFSoupSpec newSoupSpec:soupName withFeatures:nil];
+    }
     NSArray *indexSpecs = [SFSoupIndex asArraySoupIndexes:[argsDict nonNullObjectForKey:kIndexesArg]];
-    [self log:SFLogLevelDebug format:@"registerSoup with name: %@, indexSpecs: %@", soupName, indexSpecs];
-    
-    NSError* error = nil;
-    BOOL regOk = [[self getStoreInst:argsDict] registerSoup:soupName withIndexSpecs:indexSpecs error:&error];
-    if (regOk) {
-        callback(@[[NSNull null], soupName]);
+    [self log:SFLogLevelDebug format:@"registerSoup with name: %@, soup features: %@, indexSpecs: %@", soupSpec.soupName, soupSpec.features, indexSpecs];
+    if (smartStore) {
+        NSError *error = nil;
+        BOOL result = [smartStore registerSoupWithSpec:soupSpec withIndexSpecs:indexSpecs error:&error];
+        if (result) {
+            callback(@[[NSNull null], soupName]);
+        } else {
+            callback(@[RCTMakeError(@"registerSoup failed", error, nil)]);
+        }
     } else {
-        callback(@[RCTMakeError(@"registerSoup failed", error, nil)]);
+        callback(@[RCTMakeError(@"registerSoup failed", nil, nil)]);
```

Comment:
Not passing error anymore? If smartstore is nil, then we should somehow reflect that in the error.